packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -153,7 +153,16 @@ const helpers = {
           );
         } else {
           logger.warn(`Unable to find a content for ${mediaTypes}`);
-          return left(ProblemJsonError.fromTemplate(NOT_ACCEPTABLE, `Unable to find content for ${mediaTypes}`));
+
+          if (_httpOperation.method === 'head') {
```

Comment:
Hmm, I think we might want to make this check as soon as possible so we can skip the whole negotiation process if possible?

In this case though, the response body will always be empty even if the user has effectively specified a content for the HEAD verb. @philsturgeon what do you think? Maybe Spectral should somehow notice this?

Reply:
@XVincentX, should we do the re-write here? Moving up the check would require changes in finding the `response`

Reply:
When in doubt, the HTTP specification is a good baseline for what how to handle most situations. RFC 7231 says this about HEAD:

```
4.3.2.  HEAD

   The HEAD method is identical to GET except that the server MUST NOT
   send a message body in the response (i.e., the response terminates at
   the end of the header section).  The server SHOULD send the same
   header fields in response to a HEAD request as it would have sent if
   the request had been a GET, except that the payload header fields
   (Section 3.3) MAY be omitted.  This method can be used for obtaining
   metadata about the selected representation without transferring the
   representation data and is often used for testing hypertext links for
   validity, accessibility, and recent modification.

   A payload within a HEAD request message has no defined semantics;
   sending a payload body on a HEAD request might cause some existing
   implementations to reject the request.

   The response to a HEAD request is cacheable; a cache MAY use it to
   satisfy subsequent HEAD requests unless otherwise indicated by the
   Cache-Control header field (Section 5.2 of [RFC7234]).  A HEAD
   response might also have an effect on previously cached responses to
   GET; see Section 4.3.5 of [RFC7234].
```

The important bit here is "The HEAD method is identical to GET except that the server MUST NOT send a message body in the response". If they have defined a response we don't need to show it. 

**Personal opinion:** Maybe 1 user ever is going to define a head with a body, then when it doesn't work they'll google around and find out they should not have done that.

**Future feature idea:** if no HEAD is defined and they run a GET then it should just use the GET header content and strip the body. Lots of web application frameworks offer this as functionality too.

Reply:
@philsturgeon, is this OK ?
>Future feature idea: if no HEAD is defined and they run a GET then it should just use the GET header content and strip the body. Lots of web application frameworks offer this as functionality too.

Reply:
Is what ok?

Reply:
@philsturgeon, is what you written OK? You said `and they run a GET then it should just use the GET header content and strip the body`. Why should I strip anything if I sand GET? I'm coooooonfused

Reply:
Noted. So when an HEAD request comes In we'll skip the whole negotiation process.

Reply:
@XVincentX, so I'm moving this back to in-progress and will skip the negotiation in this PR

Reply:
Yes please. If you're in trouble or overwhelmed with other stuff please let me know, I'm now back 100% and I can share some of the work.

Reply:
I will if I do, thanks :+1:

Reply:
> @philsturgeon, is what you written OK? You said `and they run a GET then it should just use the GET header content and strip the body`. Why should I strip anything if I sand GET? I'm coooooonfused

@philsturgeon  ?

Reply:
Ignore this. @lag-of-death this is a future feature we won't implement right now.

Reply:
@XVincentX, but what does it mean? I send GET and I strip the response body? I don't get it

Reply:
@lag-of-death No I think @philsturgeon's idea is that if you do not specify a response for an HEAD request explicitly in the OpenAPI document, Prism could intelligently just take the GET request (if defined), strip its body and return it to the client.

We can skip this for now. Does that clarify?

Reply:
@XVincentX, this would make sense, yes. But I have a feeling this is not what @philsturgeon said. Maybe that's what he had in mind? I just want to be sure, is all (to be on the same page and not to make wrong assumptions).

Reply:
Yep I mentioned it as a future idea if you want to take note of it in another issue or if its easy to do now, whatever. Y'all are free to use your own judgement and manage more of the feature planning yourself.